ci/src/cI_github_hooks.ml

```diff
@@ -1,293 +1,56 @@
-      update "description" descr >>= fun () ->
-      begin match target_url with
-        | None -> ensure_removed t (dir / "target_url")
-        | Some url -> update "target_url" (Uri.to_string url)
-      end >>= fun () ->
-      let message = Fmt.strf "Set state of %a to %a" Commit.pp commit CI_state.pp_status status in
-      Log.debug (fun f -> f "set_state: %s" message);
-      DK.Transaction.commit t ~message
-    ) >>*= Lwt.return
-
-let pr snapshot id =
-  let id_leaf = string_of_int id in
-  read_string snapshot (prs_dir / id_leaf / "head") >>= function
-  | Error _ -> Log.err (fun f -> f "Missing head for PR %d" id); Lwt.return None
-  | Ok commit ->
-    let commit = { Commit.hash = String.trim commit; snapshot } in
-    read_string snapshot (prs_dir / id_leaf / "title") >|= fun title ->
-    let title =
-      match title with
-      | Error (`Msg m) ->
-        Log.err (fun f -> f "Bad title for PR %d: %s" id m);
-        Fmt.strf "Bad title: %s" m
-      | Ok t -> String.trim t in
-    Some { PR.id; commit; title }
-
-let read_opt_file snapshot path =
-  read_file snapshot path >|= function
-  | Ok data -> Some data
-  | Error (`Msg "No such file or directory") -> None
-  | Error e -> failf "Error reading %a: %a" Datakit_path.pp path DK.pp_error e
-
-let read_opt_dir {project_id; root} path =
-  let path = CI_projectID.path project_id /@ path in
-  DK.Tree.read_dir root path >|= function
-  | Ok items -> items
-  | Error (`Msg "No such file or directory") -> []
-  | Error e -> failf "Error reading %a: %a" Datakit_path.pp path DK.pp_error e
-
-let prs snapshot =
-  read_opt_dir snapshot prs_dir >>=
-  Lwt_list.fold_left_s (fun acc id ->
-      match String.to_int id with
-      | None -> Log.warn (fun f -> f "Invalid PR ID %S" id); Lwt.return acc
-      | Some id -> pr snapshot id >|= function
-        | None -> acc
-        | Some value -> IntMap.add id value acc
-    ) IntMap.empty
-
-let refs snapshot =
-  let open! Datakit_path.Infix in
-  let results = ref Datakit_path.Map.empty in
-  let rec scan ~context leaf =
-    let context = context / leaf in
-    read_opt_file snapshot (refs_dir /@ context / "head") >>= function
-    | Some head ->
-      let hash = String.trim (Cstruct.to_string head) in
-      let head = { Commit.snapshot; hash } in
-      results := Datakit_path.Map.add context { Ref.head; name = context } !results;
-      Lwt.return ()
-    | None ->
-      read_opt_dir snapshot (refs_dir /@ context) >>=
-      Lwt_list.iter_s (scan ~context)
-  in
-  read_opt_dir snapshot refs_dir >>=
-  Lwt_list.iter_s (scan ~context:Datakit_path.empty) >|= fun () ->
-  !results
-
-module Target = struct
-  type t = [ `PR of PR.t | `Ref of Ref.t ]
-
-  let dispatch p r = function
-    | `PR x -> p x
-    | `Ref x -> r x
-
-  let head = dispatch PR.head Ref.head
+  type t = string list
+  let circle_ci = ["ci"; "circleci"]
+  let datakit_ci x = "ci" :: "datakit" :: String.cuts ~sep:"/" x
```

Comment:
I will revert that, that change got intertwined with the big rewrite, which is bad.